bindings/python/edlib.pyx

```diff
@@ -34,10 +35,44 @@ def align(query, target, mode="NW", task="distance", k=-1, additionalEqualities=
                 Match: '=', Insertion to target: 'I', Deletion from target: 'D', Mismatch: 'X'.
                 e.g. cigar of "5=1X1=1I" means "5 matches, 1 mismatch, 1 match, 1 insertion (to target)".
     """
-    # Transform python strings into c strings.
-    cdef bytes query_bytes = query.encode('utf-8') if type(query) != bytes else query;
+    # Transform python sequences of hashables into c strings.
+    cdef bytes query_bytes
+    cdef bytes target_bytes
+    query_needs_alphabet = False
+    target_needs_alphabet = False
+    if isinstance(query, bytes):
+        query_bytes = query
+    elif isinstance(query, str):
+        query_bytes = query.encode('utf-8')
+        query_needs_alphabet = len(query_bytes) != len(query)
+    else:
+        query_needs_alphabet = True
+
+    if isinstance(target, bytes):
+        target_bytes = target
+    elif isinstance(target, str):
+        target_bytes = target.encode('utf-8')
+        target_needs_alphabet = len(target_bytes) != len(target)
+    else:
+        target_needs_alphabet = True
+
+    # Map non-ascii symbols into an ASCII alphabet so it can be used
+    # in the C++ code
+    alphabet = {}
```

Comment:
Maybe name this one more descriptively, for example `inputToByteMapping` or smth like that, since this is a mapping actually from one space to another.